sv_inline.h

```diff
@@ -770,48 +770,68 @@ Perl_SvPADSTALE_off(SV *sv)
 
 /*
 =for apidoc_section $SV
-=for apidoc         SvIV
-=for apidoc_item    SvIV_nomg
-=for apidoc_item m||SvIVx
-
-These each coerce the given SV to IV and return it.  The returned value in many
-circumstances will get stored in C<sv>'s IV slot, but not in all cases.  (Use
-C<L</sv_setiv>> to make sure it does).
-
-As of 5.37.1, all are guaranteed to evaluate C<sv> only once.
-
-C<SvIVx> is now identical to C<SvIV>, but prior to 5.37.1, it was the only form
-guaranteed to evaluate C<sv> only once.
-
-C<SvIV_nomg> is the same as C<SvIV>, but does not perform 'get' magic.
-
-=for apidoc         SvNV
-=for apidoc_item    SvNV_nomg
-=for apidoc_item m||SvNVx
-
-These each coerce the given SV to NV and return it.  The returned value in many
-circumstances will get stored in C<sv>'s NV slot, but not in all cases.  (Use
-C<L</sv_setnv>> to make sure it does).
-
-As of 5.37.1, all are guaranteed to evaluate C<sv> only once.
-
-C<SvNVx> is now identical to C<SvNV>, but prior to 5.37.1, it was the only form
-guaranteed to evaluate C<sv> only once.
-
-C<SvNV_nomg> is the same as C<SvNV>, but does not perform 'get' magic.
-
-=for apidoc         SvUV
-=for apidoc_item    SvUV_nomg
-=for apidoc_item m||SvUVx
-
-These each coerce the given SV to UV and return it.  The returned value in many
-circumstances will get stored in C<sv>'s UV slot, but not in all cases.  (Use
-C<L</sv_setuv>> to make sure it does).
-
-As of 5.37.1, all are guaranteed to evaluate C<sv> only once.
-
-C<SvUVx> is now identical to C<SvUV>, but prior to 5.37.1, it was the only form
-guaranteed to evaluate C<sv> only once.
+=for apidoc          SvIV
+=for apidoc_item     SvIV_nomg
+=for apidoc_item  m||SvIVx
+=for apidoc_item  m||SvIVX
+=for apidoc_item mD||SvIVXx
+
+These each return the IV contained in the given SV.
+
+C<SvIVX> returns the raw value in the SV's IV slot, without checks or
+conversions.  Only use when you are sure C<SvIOK> is true.
+C<SvIVXx> is a deprecated synonym for C<SvIVX>.
```

Comment:
Why are and when were the `x` suffix names deprecated? (and was that perldeltaed?)

They still the right thing in older perls and harmlessly do the right thing in newer perls.

Reply:
It appears that only `SvUVXx` had been deprecated, contrary to my recollection.  And it happened in 

> commit b7822a6dc596b40a42a5d599451fa677a9dfb98a
 Author: Karl Williamson <khw@cpan.org>
 Date:   Sun Aug 4 18:03:03 2019 -0600

I don't remember why.  But they certainly don't have to be deprecated.  It turns out they were added in

> commit 463ee0b2acbd047c27e8b5393cdd8398881824c5
 Author: Larry Wall <larry@wall.org>
 Date:   Wed Nov 10 00:00:00 1993 +0000
  
     perl 5.0 alpha 4

and remained pretty much untouched since.

I've never seen them used.  Not that these are not the same as SvIVx, which was the form that untill 5.37 was the only one that guaranteed single evaluation